catalog.bom

```diff
@@ -138,102 +118,416 @@ brooklyn.catalog:
+                echo "replication:
+                   oplogSizeMB: $OPLOGSIZE
+                   replSetName: $REPLSETNAME
+                   secondaryIndexPrefetch: $SECINDPRE
+                   enableMajorityReadConcern: $ENAMAJREADCONCERN" | sudo tee -a /etc/mongod.conf
+
+                sudo service mongod restart
+                mongo --host $PRIMARYNODE --eval "rs.add('$MONGOIP:$MONGOPORT')"
+
+            shell.env:
+              PRIMARYNODE: $brooklyn:component("first-node").attributeWhenReady("mongodb.rs.primaryNode")
+              REPLSETNAME: $brooklyn:config("mongodb.rs.name")
+              OPLOGSIZE: $brooklyn:config("mongodb.rs.oplogsize")
+              SECINDPRE: $brooklyn:config("mongodb.rs.secindpre")
+              ENAMAJREADCONCERN: $brooklyn:config("mongodb.rs.enamajreadconcern")
+
+
+    - id: mongodb-replicaset-template
+      description: |
+        A cluster of MongoDB Nodes that are joined together in a Replica Set
+        allowing for data replication across multiple nodes.
+      name: "MongoDB Replica Set"
+      iconUrl: https://webassets.mongodb.com/_com_assets/cms/mongodb-logo-rgb-j6w271g1xn.jpg
+      itemType: template
+      item:
+        services:
+          - type: mongodb-replicaset-application
+            name: "MongoDB Replica Set"
+
+    - id: mongodb-server-template
+      description: |
+        A single MongoDB Node to be used as a nosql database
+      name: "MongoDB Server"
+      iconUrl: https://webassets.mongodb.com/_com_assets/cms/mongodb-logo-rgb-j6w271g1xn.jpg
+      itemType: template
+      item:
+          services:
+            - type: mongodb-server-application
+              name: "MongoDB Server"
+
+
+    - id: mongodb-replicaset-application
+      description: |
+        A cluster of MongoDB Nodes that are joined together in a Replica Set
+        allowing for data replication across multiple nodes.
+      name: "MongoDB Replica Set"
+      iconUrl: https://webassets.mongodb.com/_com_assets/cms/mongodb-logo-rgb-j6w271g1xn.jpg
+      itemType: entity
+      item: 
+        type: org.apache.brooklyn.entity.stock.BasicApplication
+
+        brooklyn.parameters:
+          - name: mongodb.rs.name
+            type: string
+            label: Replica Set name
+            description: |
+              The name of the Replica Set
+            default: ReplicaSetName
+
+          - name: mongodb.rs.oplogsize
+            type: string
+            label: Operation Log Size
+            description: |
+              The Size of the operation log in Megabytes
+            default: 4096
+
+          - name: mongodb.rs.secindpre
+            type: string
+            label: Secondary Index Prefetch
+            description: |
+              Indexes that a secondary member will load into memory before applying 
+              operations from oplog 
+            default: all
+
+          - name: mongodb.rs.enamajreadconcern
+            type: boolean
```

Comment:
This actually *works* when running against the latest Brooklyn snapshot compiled from source.